### PR TITLE
Feat/breakout cache main session locus

### PR DESCRIPTION
**Pull request by othellashao**
# COMPLETES # 
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-423972

## This pull request addresses

Maintain a main session locus cache, and replace it when host/cohost leave from bo since locus won't send the full data on this case

## by making the following changes

1. add main session locus cache
2. determine whether to clear the cache when user join to bo by main locus url (follow native client's logic)
3. when user return to main session, check to use the cache or not
4. fix some issues 

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [x] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/plugin-meetings/test/unit/spec/locus-info/index.js [hunk 1]
```diff
@@ -1553,10 +1553,145 @@ describe('plugin-meetings', () => {
...
+        const result = locusInfo.mergeParticipants(participants, sourceParticipants);
+        assert.deepEqual(result, [{id: '111', status: 'LEFT'}, {id: '222'}, {id: '333'}]);
+      });
+
+      it('return new participants if previous participants is empty', () => {
+        const result = locusInfo.mergeParticipants([], sourceParticipants);
+        assert.deepEqual(result, sourceParticipants);
+      });
+
+      it('return previous participants if new participants is empty', () => {
```
Comment (Coread) [inline packages/@webex/plugin-meetings/test/unit/spec/locus-info/index.js:1654]: issue: missing test case for when new participants is null/undefined

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js [hunk 2]
```diff
@@ -4632,6 +4632,30 @@ describe('plugin-meetings', () => {
...
+          meeting.meetingRequest.getLocusStatusByUrl = sinon.stub().returns(Promise.resolve());
+
+          await meeting.locusInfo.emit(
+            {function: 'test', file: 'test'},
+            'CONTROLS_JOIN_BREAKOUT_FROM_MAIN',
+            {mainLocusUrl}
+          );
+
+          assert.calledOnceWithExactly(meeting.meetingRequest.getLocusStatusByUrl, mainLocusUrl);
+          const error = {statusCode: 403};
```
Comment (Coread) [inline packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js:4648]: issue: There should really be a test for when the error isn't a 403 and when the error doesn't have a status code

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/request.js [hunk 3]
```diff
@@ -377,4 +377,16 @@ describe('plugin-meetings', () => {
       });
     });
   });
+
+  describe('#getLocusStatusByUrl', () => {
+    it('check lcous status', async () => {
```
Comment (Coread) [inline packages/@webex/plugin-meetings/test/unit/spec/meeting/request.js:382]: nitpick: typo of locus

**Review by Coread — CHANGES_REQUESTED** [review]
please see comments regarding additional tests

File: packages/@webex/plugin-meetings/test/unit/spec/locus-info/index.js [hunk 4]
```diff
@@ -1553,10 +1553,145 @@ describe('plugin-meetings', () => {
...
+        participants = [{id: '111', status: 'JOINED'}, {id: '222'}];
+        sourceParticipants = [{id: '111', status: 'LEFT'}, {id: '333'}];
+      });
+
+      it('merge the participants, replace it by id if exist in old array', () => {
+        const result = locusInfo.mergeParticipants(participants, sourceParticipants);
+        assert.deepEqual(result, [{id: '111', status: 'LEFT'}, {id: '222'}, {id: '333'}]);
+      });
+
+      it('return new participants if previous participants is empty', () => {
```
[the remark this answers is not in the document]
Comment (othellashao) [reply packages/@webex/plugin-meetings/test/unit/spec/locus-info/index.js:1649]: Done

File: packages/@webex/plugin-meetings/test/unit/spec/locus-info/index.js (hunk 1 above)
Comment (othellashao) [reply packages/@webex/plugin-meetings/test/unit/spec/locus-info/index.js:1662]: Done

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js (hunk 2 above)
Comment (othellashao) [reply packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js:4648]: Done

**Review by Coread — APPROVED** [review]
